src/main/java/org/opentripplanner/updater/GraphUpdaterManager.java

```diff
@@ -146,6 +145,36 @@ public void execute(GraphWriterRunnable runnable) {
         });
     }
 
+    /**
+     * This is another method to use to modify the graph from the updaters. It behaves like execute,
+     * but blocks until the runnable has been executed. This might be particularly useful in the
+     * setup method of an updater.
+     *
+     * @param runnable is a graph writer runnable
+     * @throws ExecutionException
+     * @throws InterruptedException
+     * @see GraphUpdaterManager.execute
+     */
+    public void executeBlocking(GraphWriterRunnable runnable) throws InterruptedException,
+            ExecutionException {
+        Future<?> future = executeReturningFuture(runnable);
+        // Ask for result of future. Will block and return null when runnable is successfully
+        // finished, throws otherwise
+        future.get();
+    }
+
+    private Future<?> executeReturningFuture(final GraphWriterRunnable runnable) {
+        Future<?> future = scheduler.submit(() -> {
+            try {
+                runnable.run(graph);
+            } catch (Exception e) {
+                LOG.error("Error while running graph writer {}:", runnable.getClass().getName(),
+                        e);
+            }
+        });
+        return future;
+    }
+
```

Comment:
I refactored the "more data" loop for clarity, and to replace tail recursion with a do-while loop (which should be more resilient to bad input data). The tasks don't need to be executed in a blocking way since they are all enqueued by a single thread so should never be executed out of order. I added comments explaining this to the GraphUpdaterManager methods.